### ISSUE TITLE
Question about Conv layer masking.

### ISSUE THREAD

**Issue by placebokkk** [SeanNaren/deepspeech.pytorch#491]
In model.py, it looks like all layers use the same T mask. 
```
        for module in self.seq_module:
            x = module(x)
            mask = torch.BoolTensor(x.size()).fill_(0)
            if x.is_cuda:
                mask = mask.cuda()
            for i, length in enumerate(lengths):
                length = length.item()
                if (mask[i].size(2) - length) > 0:
                    mask[i].narrow(2, length, mask[i].size(2) - length).fill_(1)
            x = x.masked_fill(mask, 0)
        return x, lengths
```

And this mask is calculated by the length (size T) of the last layer.
```
    def get_seq_lens(self, input_length):
        seq_len = input_length
        for m in self.conv.modules():
            if type(m) == nn.modules.conv.Conv2d:
                seq_len = ((seq_len + 2 * m.padding[1] - m.dilation[1] * (m.kernel_size[1] - 1) - 1) / m.stride[1] + 1)
        return seq_len.int()
```

In deepspeech2 default setting , only first Conv2d use stride=2 in T. So above method (set all layer using the same mask)works well .
```
        self.conv = MaskConv(nn.Sequential(
            nn.Conv2d(1, 32, kernel_size=(41, 11), stride=(2, 2), padding=(20, 5)),
            nn.BatchNorm2d(32),
            nn.Hardtanh(0, 20, inplace=True),
            nn.Conv2d(32, 32, kernel_size=(21, 11), stride=(2, 1), padding=(10, 5)),
            nn.BatchNorm2d(32),
            nn.Hardtanh(0, 20, inplace=True)
        ))    def forward(self, x, lengths):
```

But if I add another Conv2d layer with stride=2 for T,  will above code still work?  Should I need to use different masks for different layers?

**Comment by SeanNaren** [comment]
`def get_seq_lens(self, input_length)` should correctly calculate the sequence length based on all conv layers, so it should work

**Comment by placebokkk** [comment]
> `def get_seq_lens(self, input_length)` should correctly calculate the sequence length based on all conv layers, so it should work

@SeanNaren Thank you very much for quick reply.

Please let me give an example to make my point clear, When I change the second Conv2d layer using stride=2 for T as below:

```
>>> conv=nn.Sequential(
...             nn.Conv2d(1, 32, kernel_size=(41, 11), stride=(2, 2), padding=(20, 5)),
...             nn.BatchNorm2d(32),
...             nn.Hardtanh(0, 20, inplace=True),
...             nn.Conv2d(32, 32, kernel_size=(21, 11), stride=(2, 2), padding=(10, 5)),
...             nn.BatchNorm2d(32),
...             nn.Hardtanh(0, 20, inplace=True)
...         )
>>> a=torch.FloatTensor(torch.Size((3,1,80,100)))
>>> b=conv(a)
>>> b.size()
torch.Size([3, 32, 20, 25])
>>> x = a
>>> for module in conv:
...     x=module(x)
...     x.size()
...
torch.Size([3, 32, 40, 50])
torch.Size([3, 32, 40, 50])
torch.Size([3, 32, 40, 50])
torch.Size([3, 32, 20, 25])
torch.Size([3, 32, 20, 25])
torch.Size([3, 32, 20, 25])
```
The output lengths of first nn.Conv2d/nn.BatchNorm2d/nn.Hardtanh are **50** , while the second are **25**.

Assuming we have three samples in one batch, all with 100 length. Now get_seq_lens return only 25.
```
>>> seq_len=torch.FloatTensor([100,100,100])
>>> for m in conv.modules():
...     if type(m) == nn.modules.conv.Conv2d:
...         seq_len = ((seq_len + 2 * m.padding[1] - m.dilation[1] * (m.kernel_size[1] - 1) - 1) / m.stride[1] + 1)
...
>>> seq_len.int()
tensor([25, 25, 25], dtype=torch.int32)
```

Then **25** will be used as mask length for all the six layers. Does this right? Why here not using **50** the first three module?
